src/index.ts

```diff
@@ -86,10 +86,14 @@ function VitePluginRestart(options: Options = {}): Plugin {
     configResolved(config) {
       if (fs.existsSync('vite.config.ts'))
         configFile = 'vite.config.ts'
+      
+      // famous last words, but this *appears* to always be an absolute path
+      // with all slashes normalized to forward slashes `/`. this is compatible
+      // with path.posix.join, so we can use it to make an absolute path glob
       root = config.root
-
-      reloadGlobs = toArray(options.reload).map(i => pathPlatform.resolve(root, i))
-      restartGlobs = toArray(options.restart).map(i => pathPlatform.resolve(root, i))
+      
+      restartGlobs = toArray(options.restart).map(i => path.posix.join(root, i))
+      reloadGlobs = toArray(options.reload).map(i => path.posix.join(root, i))
```

Comment:
Join is not ideal as relative paths like `../../` should be supported.

Reply:
oh, sorry. Just saw this comment after the break.

I *thought* that `path.posix.join` supported normalizing relative paths. 
![image](https://user-images.githubusercontent.com/4720546/148169510-4f602d4e-2182-4f9e-b9dd-e640f3eb9813.png)